src/components/modal/modal.js

```diff
@@ -29,31 +29,57 @@ const Header = styled.div`
 `;
 
 class Modal extends React.Component {
+  static propTypes = {
+    onToggle: PropTypes.func,
+    isOpen: PropTypes.bool,
+  };
+
   static defaultProps = {
-    isOpened: false,
+    isOpen: false,
   };
 
   state = {
-    isOpened: this.props.isOpened,
+    isOpen: this.props.isOpen,
   };
 
-  toggle = () => this.setState({ isOpened: !this.state.isOpened });
   handleClickOutside = (e) => {
-    e.target.id === this.props.id ? this.toggle() : this.setState({ isOpened: false });
+    const { onToggle } = this.props;
+
+    if (!this.state.isOpen) {
+      return;
+    }
+
+    e.stopPropagation();
+
+    this.setState({ isOpen: false });
```

Comment:
this.setState({ isOpen });   
 if (onToggle) {
      onToggle({ isOpen });
    }

can be extracted in a function, cause is used a few times